conformance/runner.js

```diff
@@ -161,8 +158,12 @@ function runTest(spec) {
   const updateTest = function(spec) {
     firestore.api.Firestore._commit = function(request) {
       const expected = convertCommit(spec.request);
-      assert.deepEqual(request, expected);
-      resolve();
+      try {
+        assert.deepEqual(request, expected);
+        resolve();
```

Comment:
I reworked this to take advantage of the error handling that is built into the SDK. This means that I now have to return proper responses from the request handlers, but I moved all this logic to a single place.

I added the try/catch originally to handle unhandled promise rejections. The asserts already failed the tests, but returned unhandled Promised back to Node. Since everything is wrapped in Promise.resolve() and request blocks now, this special handling is no longer necessary.